### ISSUE TITLE
2GB VRAM spike after generating images?

### ISSUE THREAD

**Issue by TheEnhas** [hlky/stable-diffusion#77]
Hello, I'm been trying out this fork/UI with the --optimized-turbo parameter and it's been working fine, but after generating I tend to get this every single time:

CUDA out of memory. Tried to allocate 2.00 GiB (GPU 0; 8.00 GiB total capacity; 5.29 GiB already allocated; 282.00 MiB free; 5.37 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

This was with a single batch of 8 512x512 images. It's not the generation itself that leads to OOM, but whatever this is, which also prevents the images from being saved. In contrast I can generate 20 512x512 images with basujindal's optimized fork and I don't get the VRAM spike at the end, so something's definitely off here.

**Comment by hlky** [comment]
Are you trying to use gfpgan or esrgan as well?

**Comment by TheEnhas** [comment]
No, not during generation time. I also tried adding the --extra-models-cpu parameter (thinking maybe it's trying to load those into VRAM even if I'm not using them) but it also made no difference.

**Comment by hlky** [comment]
Can you try removing the esrgan and gfpgan model files?

Might be somewhere that needs torch_gc() calling to clear cache

**Comment by TheEnhas** [comment]
Removed the model files, still got the 2GB VRAM spike and OOM error. So it's not that.

**Comment by hlky** [comment]
Thanks, moving to dev repo to get more attention of this issue.
